Modules/Core/Common/include/itkBufferedImageNeighborhoodPixelAccessPolicy.h

```diff
@@ -0,0 +1,137 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef itkBufferedImageNeighborhoodPixelAccessPolicy_h
+#define itkBufferedImageNeighborhoodPixelAccessPolicy_h
+
+#include "itkIndex.h"
+#include "itkOffset.h"
+#include "itkSize.h"
+
+#include <cassert>
+
+namespace itk
+{
+namespace Experimental
+{
+
+/**
+ * \class BufferedImageNeighborhoodPixelAccessPolicy
+ * ImageNeighborhoodPixelAccessPolicy class for ShapedImageNeighborhoodRange.
+ * Allows getting and setting the value of a pixel, located in a specified
+ * neighborhood location, at a specified offset. Assumes that the pixel index
+ * is within the buffered region.
+ *
+ * \note This policy offers faster pixel access than other pixel access policies,
+ * because it does not do any bounds checking or border extrapolation. An
+ * attempt to access pixels outside the buffered region will have undefined behavior.
+ *
+ * \author Niels Dekker, LKEB, Leiden University Medical Center
+ *
+ * \see ShapedNeighborhoodIterator
+ * \see ConstantBoundaryImageNeighborhoodPixelAccessPolicy
+ * \see ZeroFluxNeumannImageNeighborhoodPixelAccessPolicy
+ * \ingroup ImageIterators
+ * \ingroup ITKCommon
+ */
+template <typename TImage>
+class BufferedImageNeighborhoodPixelAccessPolicy final
+{
+private:
+  using NeighborhoodAccessorFunctorType = typename TImage::NeighborhoodAccessorFunctorType;
+  using PixelType = typename TImage::PixelType;
+  using InternalPixelType = typename TImage::InternalPixelType;
+
+  using ImageDimensionType = typename TImage::ImageDimensionType;
+  static constexpr ImageDimensionType ImageDimension = TImage::ImageDimension;
+
+  using IndexType = Index<ImageDimension>;
+  using OffsetType = Offset<ImageDimension>;
+  using ImageSizeType = Size<ImageDimension>;
+
+  // Index value to the image buffer, indexing the current pixel.
+  const IndexValueType m_PixelIndexValue;
+
+  // A reference to the accessor of the image.
+  const NeighborhoodAccessorFunctorType & m_NeighborhoodAccessor;
+
+
+  // Private helper function. Calculates and returns the index value of the
+  // current pixel within the image buffer.
+  static IndexValueType
+  CalculatePixelIndexValue(const ImageSizeType &
+#if !defined(NDEBUG)
+                             // Parameter that is only needed for debug assert.
+                             imageSize
+#endif
+                           ,
+                           const OffsetType & offsetTable,
+                           const IndexType &  pixelIndex) ITK_NOEXCEPT
+  {
+    IndexValueType result = 0;
+
+    for (ImageDimensionType i = 0; i < ImageDimension; ++i)
+    {
+      const auto pixelIndexValue = pixelIndex[i];
+      assert((pixelIndexValue >= 0) && (static_cast<SizeValueType>(pixelIndexValue) < imageSize[i]));
```

Comment:
This assertion gets triggered 😄
https://open.cdash.org/testDetails.php?test=819819335&build=6195188